tests/security/authn_policy/scripts/authn_policy.sh

```diff
@@ -0,0 +1,97 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -e
+set -u
+set -o pipefail
+
+source "${REPO_ROOT}/content/en/docs/tasks/security/authentication/authn-policy/snips.sh"
+source "${REPO_ROOT}/tests/util/samples.sh"
+
+snip_setup_1
+
+sample_wait_for_deployment foo httpbin
+sample_wait_for_deployment foo sleep
+sample_wait_for_deployment bar httpbin
+sample_wait_for_deployment bar sleep
+sample_wait_for_deployment legacy httpbin
+sample_wait_for_deployment legacy sleep
+
+_run_and_verify_same  snip_setup_2 "$snip_setup_2_out"
+_run_and_verify_same  snip_setup_3 "$snip_setup_3_out"
+_run_and_verify_same  snip_setup_4 "$snip_setup_4_out"
+_run_and_verify_same  snip_setup_5 "$snip_setup_5_out"
+
+_run_and_verify_like  snip_auto_mutual_tls_1 "$snip_auto_mutual_tls_1_out"
+_run_and_verify_same  snip_auto_mutual_tls_2 "$snip_auto_mutual_tls_2_out"
+
+snip_globally_enabling_istio_mutual_tls_in_strict_mode_1
+
+_run_and_verify_same  snip_globally_enabling_istio_mutual_tls_in_strict_mode_2 "$snip_globally_enabling_istio_mutual_tls_in_strict_mode_2_out"
+
+snip_cleanup_part_1_1
+
+snip_namespacewide_policy_1
+
+_run_and_verify_same  snip_namespacewide_policy_2 "$snip_namespacewide_policy_2_out"
+
+snip_enable_mutual_tls_per_workload_1
+snip_enable_mutual_tls_per_workload_2
+
+_run_and_verify_same  snip_enable_mutual_tls_per_workload_3 "$snip_enable_mutual_tls_per_workload_3_out"
+
+# Ignore snip_enable_mutual_tls_per_workload_4()--it's just text.
+
+snip_enable_mutual_tls_per_workload_5
+snip_enable_mutual_tls_per_workload_6
+
+_run_and_verify_same  snip_enable_mutual_tls_per_workload_7 "$snip_enable_mutual_tls_per_workload_7_out"
+
+snip_policy_precedence_1
+snip_policy_precedence_2
+
+_run_and_verify_same  snip_policy_precedence_3 "$snip_policy_precedence_3_out"
+
+snip_cleanup_part_2_1
+
+snip_enduser_authentication_1
+snip_enduser_authentication_2
+
+# Export the INGRESS_ environment variables
+sample_set_ingress_environment_variables
+
+_run_and_verify_same  snip_enduser_authentication_3 "$snip_enduser_authentication_3_out"
+
+snip_enduser_authentication_4
+
+_run_and_verify_same  snip_enduser_authentication_5 "$snip_enduser_authentication_5_out"
+_run_and_verify_same  snip_enduser_authentication_6 "$snip_enduser_authentication_6_out"
+_run_and_verify_same  snip_enduser_authentication_7 "$snip_enduser_authentication_7_out"
+
+snip_enduser_authentication_8
+snip_enduser_authentication_9
+
+_run_and_verify_same  snip_enduser_authentication_10 "$snip_enduser_authentication_10_out"
```

Comment:
If we make `__verify_with_retry` public, tests could pass in a custom cmp function. Then we could use the generated _out snippet here, otherwise I suppose using verify_contains is the best you can do.
```suggestion
expected="200
400"
_run_and_verify_contains snip_enduser_authentication_10 $expected
```

Reply:
Thanks for the great suggestion for the fix.  You saved me some time noodling on how to address the issue--all my immediate solutions were needlessly complex.